### ISSUE TITLE
Multicurrency calculates on corporate currency and not record currency. Issues arises with rounding values after applying exchange rate.

### ISSUE THREAD

**Issue by patvachon77** [afawcett/declarative-lookup-rollup-summaries#284]
Described in this video.
https://www.youtube.com/watch?v=z5ylMzp9tBc

**Comment by afawcett** [comment]
Yeah so i've been round the loop on this before with Salesforce support, it has to do with how SOQL Aggregate functions work over Currency fields. If the tool where to read the records itself and do the additions in memory it would arrive at the correct value, however this is obviously less efficient data volumes wise. 

Here is some other GitHub issues, https://github.com/afawcett/declarative-lookup-rollup-summaries/issues/217, https://github.com/afawcett/declarative-lookup-rollup-summaries/issues/57

**Comment by afawcett** [comment]
This is [post](https://developer.salesforce.com/forums/?id=906F00000008y4vIAA) also gives an explanation of what SOQL Aggregate functions do with currencies and is what the tool is doing to convert to the parent current.

**Comment by afawcett** [comment]
I guess i could add an option in the rollup to have it not use SOQL Aggregate queries, retrieve all child records and add up in Apex code, but honestly i'm not sure this is really going to help much when the volumes go up and it starts causing governor and/or performance issues.

**Comment by AllisonJames** [comment]
TBH, I'm of help in how Apex does or doesn't work. What I do know is that in my situation I will only ever be dealing no more than a few records to rollup at a time, then I'm not sure there's any issue with governor and/or performance.

Can you add this other option and warn user when this option is chosen?

**Comment by afawcett** [comment]
Yep, that works, marked as enhancement!

**Comment by RobLofaro** [comment]
This seems to still be an issue in our org - and we are using the rolled up fields to produce customer facing quotes.  Anyone in the UK will have incorrect values since they are GBP and corporate is in USD.  Is there a workaround or fix?

**Comment by afawcett** [comment]
Yes, basically i need to merge this PR, https://github.com/afawcett/declarative-lookup-rollup-summaries/pull/591 do some testing and release it. I'll bump the priority of this item for my next round of work on this.

**Comment by afawcett** [comment]
See https://github.com/afawcett/declarative-lookup-rollup-summaries/issues/82

**Comment by afawcett** [comment]
Added some comments to PR before merging, https://github.com/afawcett/declarative-lookup-rollup-summaries/pull/591

**Comment by harrisoe69** [comment]
@afawcett Where is the final resolution for this?

**Comment by Fr33zm4n** [comment]
Still having this issue. Any idea at a deadline to fix this?

**Comment by afawcett** [comment]
So... i read on another issue someone say they created a Formula Currency field to shadow the physical currency field. They then used the formula field as the source of the aggregate and that resolved their issue. Thought I would share here. 👍

**Comment by afawcett** [comment]
https://github.com/afawcett/declarative-lookup-rollup-summaries/pull/591/files
